door-open-service/esp8266-nodemcu/init.lua

```diff
@@ -13,16 +13,18 @@ gpio.write(gpio_pin, gpio.LOW)
 
 -- 3. Set-up timer
 function setup_alarm()
-  tmr.alarm(0, 3000, 1, function ()
-    local ip = wifi.sta.getip()
-    if ip then
-      tmr.stop(0)
+  if wifi.sta.getip() then
+    connect_to_sf_service()
+  else
+    configure_wifi()
```

Comment:
Yeah, I'll change it for the sake of good-lookingness, although thinking of race conditions and multithreading on an ESP8266 is not very useful :P